apps/bfd-shared-utils/src/main/java/gov/cms/bfd/sharedutils/generators/token/pattern/TokenAllOf.java

```diff
@@ -0,0 +1,115 @@
+package gov.cms.bfd.sharedutils.generators.token.pattern;
+
+import gov.cms.bfd.sharedutils.generators.exceptions.GeneratorException;
+import java.math.BigInteger;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Set;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Represents an 'and' group, which is going to only really be the top level group with the current
+ * API.
+ *
+ * <p>This would represent an entire token pattern definition.
+ */
+@RequiredArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+public class TokenAllOf extends TokenPattern {
+
+  private final List<TokenPattern> patterns;
+
+  @Override
+  public boolean isValidPattern(String value) {
+    boolean isValid = true;
+
+    int startIndex = 0;
+
+    for (int i = 0; isValid && i < patterns.size(); ++i) {
+      TokenPattern tokenPattern = patterns.get(i);
+
+      String substring = value.substring(startIndex, startIndex + tokenPattern.tokenLength());
+      isValid = tokenPattern.isValidPattern(substring);
+      startIndex += tokenPattern.tokenLength();
+    }
+
+    return isValid;
```

Comment:
I think a `for (TokenPattern tokenPattern:patterns)` with a `return false` inside would be cleaner.

Reply:
It may work just fine here, I have a tendency to avoid fast fails because they are quite prone to refactoring bugs in my experience.